src/plugins/content_management/public/registry/content_type_details.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CrudClient } from '../crud_client';
+
+/**
+ * Content type definition as it is registered in the content registry.
+ */
+export interface ContentTypeDetails {
+  /**
+   * ID of the type. Must be unique. Like "dashboard", "visualization", etc.
+   */
+  id: string;
+
+  /**
+   * Human-readable name of the type. Like "Dashboard", "Visualization", etc.
+   */
+  name?: string;
+
+  /**
+   * Human-readable description of the type.
+   */
+  description?: string;
+
+  /**
+   * Icon to use for this type. Usually an EUI icon type.
+   *
+   * @see https://elastic.github.io/eui/#/display/icons
+   */
+  icon?: string;
+
+  /**
+   * Specifies item type. Defaults to 'other'.
+   */
+  kind?: ContentTypeKind;
+
+  /**
+   * CRUD client to use for this type.
+   * If not provided the default CRUD client is used assuming that this type has a server-side content registry
+   */
+  crud?: CrudClient;
+}
+
+/**
+ * Specifies whether this type represents user-like items (like user profiles)
+ * or something else (like dashboards). Defaults to 'other'.
+ */
+export type ContentTypeKind = 'user' | 'other';
```

Comment:
Do you mind explaining why we need this differentiation. Maybe we should also add a comment in the code to explain why we treat `user` content different than other type?

I also wonder why we can't just read the `type` (current `id`) which would be "user" ?

Reply:
Let's just remove this for now as this might be not needed :D